W-12739340 incorp feeback k8get started duke (#215)
mulesoft/docs-gateway: gateway/1.4/modules/ROOT/pages/flex-gateway-getting-started-kubernetes.adoc

//TODO: VERIFY Manage Servers and Read Servers permissions in RTM NO LONGER NEEDED

//os prereqs

> * A Linux or MacOS machine with an x86 processor. M1 and M2 processors are not supported for running Flex Gateway.

Author: can run flexctl (invoked when registering)  in M1 and M2

Author: removing the bullet

* Support for the creation of a `Service` resource of the `LoadBalancer` type.

> +

Author: @seblaz ok, here's the suggested update, below. Pls check it out when you get a chance.

> To avoid deployment errors when using this guide, your cluster must support this default process. However, note that Flex Gateway supports these services: `ClusterIP`, `NodePort`, `LoadBalancer` (default), `ExternalName`. 

Reviewer: I'm not sure if we can call this a "process". Also, the LoadBalancer type is the default option because we defined it so when creating the helm chart, but it's not a kubernetes default. I suggest something like this:

```suggestion
This requirement avoids deployment errors when you use this guide. However, it is possible to customize the `Service` type when deploying a gateway. Flex Gateway supports the following services: `ClusterIP`, `NodePort`, `LoadBalancer` (default), `ExternalName`. See the `service.type` key in the xref:https://artifacthub.io/packages/helm/flex-gateway/flex-gateway[Helm chart for Flex Gateway].
```

> This requirement avoids deployment errors when you use this guide. However, it is possible to customize the `Service` type when deploying a gateway. Flex Gateway supports the following services: `ClusterIP`, `NodePort`, `LoadBalancer` (default), `ExternalName`. See the `service.type` key in the https://artifacthub.io/packages/helm/flex-gateway/flex-gateway[Helm chart for Flex Gateway^].

Reviewer 2: Is the "when you use this guide" necessary?

Author: Yes, it's necessary only for the procedures in this guide. It would not be needed to avoid errors if they used another valid configuration. The procedures here are but one possible configuration, and people who read this guide need to get that message here so that they get exposed to other possible configs that we're not going to cover in this getting started guide. Those other configs are also major use cases, so we want people to be aware. Totally deliberate to tell them this so that they pause, think, and discover something that is useful to understand.

> This requirement avoids deployment errors when you use this guide. However, it is possible to customize the `Service` type when deploying a gateway. Flex Gateway supports the following services: `ClusterIP`, `NodePort`, `LoadBalancer` (default), `ExternalName`. See the `service.type` key in the xref:https://artifacthub.io/packages/helm/flex-gateway/flex-gateway[Helm chart for Flex Gateway].

Author: revised. @seblaz can you take a look at line 66?

I looked for examples in our docs but ended up pointing to the Helm chart.

Reviewer: LGTM!

Reviewer: Yes, we decided not to duplicate the docs from artifact hub here. I think it would be best to have them in our docs as well, but it seems like too much effort. Maybe sometime we will be able to sync them automatically and have them here too

To determine whether your cluster is ready for the <<deploy-flex, deployment procedure>>, review <<check-cluster>> _before starting_ the deployment. The checks require you to run a set of `kubectl` commands.

[[download-flex]]
== Step 1: Download the Flex Gateway Image

Use Docker to download the Flex Gateway image from the Docker Hub registry. 

> This procedure is a prerequisite to registering the Flex Gateway image with Anypoint Platform in <<register-flex>>. Registration is required during the gateway deployment through Helm in <<deploy-flex>>. 

Reviewer: ```suggestion
This procedure is a prerequisite to registering the Flex Gateway with Anypoint Platform in <<register-flex>>. Registration is required during the gateway deployment through Helm in <<deploy-flex>>. 
```
or
```suggestion
This procedure is a prerequisite to registering the gateway with Anypoint Platform in <<register-flex>>. Registration is required during the gateway deployment through Helm in <<deploy-flex>>. 
```

+
This command creates the `gateway` namespace and `ingress` release if they do not exist. The command syntax for the Helm repository and chart names is `<helm-repo-name>/<helm-chart-name>`. 
+

> The command uses `--set gateway.mode=connected` because the default for the Helm chart is `--set gateway.mode=local`.

Author: ```suggestion
The command uses `--set gateway.mode=connected` because the default is Local Mode. 
```

Author: done

+
If you receive an `UPGRADE FAILED` response, see <<upgrade-failed,troubleshooting>> information about this issue. 

> . If you want to check the status of the Pod that contains your gateway, run the following `kubectl` command:

Reviewer: ```suggestion
. If you want to check the status of the Pod that runs your gateway, execute the following `kubectl` command:
```

> . If you want to check the status of the Pod that runs your gateway, execute the following `kubectl` command:

Author: revised. can you check line 244 @seblaz ?

+
If you receive an `UPGRADE FAILED` response, see <<upgrade-failed,troubleshooting>> information about this issue. 

> . Check the status of your new pod by running the following `kubectl` command:

Author: ```suggestion
. Optional: Check the status of your new pod by running the following `kubectl` command:
```

+
.. Provide an API name, such as *my-api*.
.. Select *HTTP API* as the asset type.
+

> Though this guide focuses on a simple HTTP API, you can find HTTPS configuration details, see xref:flex-conn-tls-config.adoc[].

Author: move to the correct location, when you select the config

Author: moved

+
image::gateway-quickstart-k8-api-config-endpoint.png["Screenshot of 'Endpoint' page",width=90%]
+
.. For the *Scheme*, use the default, *HTTP* setting.
+
This guide focuses on a simple HTTP API. For HTTPS configuration details, see xref:flex-conn-tls-config.adoc[].
.. Copy the following URI into the *Implementation URI* field:
+
[source]
----
https://jsonplaceholder.typicode.com/
----
.. Type `80` in the *Port* field.
+

> The example in this guide uses HTTP port `80` instead of the API Manager default port, `8081`. 

Author: revised @seblaz , can you take a look at line 323

Reviewer: Can we add that the port 80 is mapped in the service? So that users are not tempted to change it leading to errors (at least when running this guide). Something like this:
```
The example in this guide uses HTTP port `80` instead of the API Manager default port, `8081`. This is because by default, the Service created maps the incoming port `80` to the target port 80 on the Pod. (and maybe we can add a link to the chart to customize this if they want)
```

> The example in this guide assumes an external IP that uses port `80` instead of the default port, `8081`. 

Reviewer: I'm not sure what the external IP refers to. Also, 8081 is the default port in APIM, but 80 is the default for the HTTP protocol, therefore I don't think we can say that 8081 is "the default port".

Author: done

> The example in this guide assumes an external IP that uses port `80` instead of the default port, `8081`. 

Author: ```suggestion
The example in this guide uses HTTP port `80` instead of the API Manager default port, `8081`. 
```

+
image::gateway-quickstart-k8-api-config-endpoint.png["Screenshot of 'Endpoint' page",width=90%]
+
.. For the *Scheme*, use the default, *HTTP* setting.
+
This guide focuses on a simple HTTP API. For HTTPS configuration details, see xref:flex-conn-tls-config.adoc[].
.. Copy the following URI into the *Implementation URI* field:
+
[source]
----
https://jsonplaceholder.typicode.com/
----
.. Type `80` in the *Port* field.

> +

Author: here's the suggested update, below, @seblaz 

please check it out when you have some time. Thx!

https://jsonplaceholder.typicode.com/
----
.. Type `80` in the *Port* field.
+

> The example in this guide assumes an external IP that uses port `80` instead of the default port, `8081`. 

Author: from seba: maybe create a separate page on all the typical things you can do when deploying Flex, such as

* node affinity
* changing port
* setting cpu and memory (prob need recommendations with x APIs and y "traffic")
* readiness (after Flex starts when Flex is ready to receive traffic)
* HA settings (downgrade and ?)
* other common ones (TBD which and how many)

Author: created W-12762970 to assess scope of future doc
